### ISSUE TITLE
False positive in Style/RedundantEach with block-pass arguments

### ISSUE THREAD

**Issue by Darhazer** [rubocop/rubocop#11137]
Given the code:
```
records.each(&:lock!).each do |record|
end
```

`Style/RedundantEach` says "Remove redundant each" on the first `each`.

Note that this passes the cop:
```
records.each { |record| record.lock! }.each do |record|
end
```
So it's only when there is a block-pass that it ignores that there is an actual block implementation.

--------

## Expected behavior

No offences

## Actual behavior

```
test.rb:2:8: C: [Corrected] Style/RedundantEach: Remove redundant each.
records.each(&:lock!).each do |record|
       ^^^^^
```

It's gets corrected to:
```
records(&:lock!).each do |record|
end
```

## RuboCop version
```
rubocop -V
1.38.0 (using Parser 3.1.2.1, rubocop-ast 1.23.0, running on ruby 3.1.2) [arm64-darwin21]
```

**Comment by Mangara** [comment]
I noticed the same with two symbols: `threads.each(&:kill).each(&:join)`, so the last doesn't need to be a block.
